src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationLiveTests.cs

```diff
@@ -504,13 +501,13 @@ public async Task GetWithLabel()
                 await service.SetAsync(s_testSetting);
 
                 // Test
-                Response<ConfigurationSetting> response = await service.GetAsync(key: s_testSetting.Key, options: s_testSetting.Label, CancellationToken.None);
+                ResponseTask<ConfigurationSetting> task = service.GetAsync(key: s_testSetting.Key, options: s_testSetting.Label, CancellationToken.None);
```

Comment:
I think this patter of `await task` is a bit un-intuitive because if requires you to know that the task will go away (along with the response object) if you await it.

I was going to suggest that we use an out parameter for the response if you want to get it, but async methods can't do that. 
And we can't overload on return type either (where we could have a method returning a tuple or just the value).

Our best bet at this point is to document this pattern then. What do you think?

Reply:
I am playing with this code a bit more (based on @stephentoub's feedback) and it's very clunky. I am actually surprised that the tests pass, and looking into why.